shared/app/nav.native.js

```diff
@@ -42,6 +49,16 @@ class CardStackShim extends Component {
     }
   }
 
+  shouldComponentUpdate(nextProps: CardStackShimProps) {
+    return (
+      this.props.mode !== nextProps.mode ||
```

Comment:
and we use this many places already (https://github.com/keybase/client/blob/7e9ad67c0f86a82649f2e81586986892adcdf6fa/shared/common-adapters/usernames.js#L75) etc